### ISSUE TITLE
Feedback: centering shapes by default in V2 seems wrong

### ISSUE THREAD

**Issue by hjfreyer** [jscad/OpenJSCAD.org#642]
I'm starting out playing with V2, and I was very surprised that `cuboid()` returns a centered shape. I understand it may be too late to change, but I'd like to argue it should revert to a corner. 

The two arguments I can think of are:

1. I've generally only used 2D drawing libraries in the past, but I've never encountered one that used the center for defining a rectangle's position. Even shapes like circles (which you might expect to be based on their center) are often specified based on their bounding box. This doesn't _itself_ mean it's the wrong choice, but certainly it means people might be surprised, and will want an easy way to instead specify a corner's location. Which brings us to the next point...
2. It's much easier to center an object than to justify ("originate"? "un-center"?) it. Case in point: the V2 library already has a "center" transformation, and doesn't have anything that moves a corner to a specific spot. And it's clear why. For any shape, there's a unique translation that moves its center to a given point. But for a cuboid, there are 8 possible translations that put a vertex on a given spot. Constructing a cuboid with a corner at (0, 0, 0) allows a user to easily choose the cuboid's orientation by specifying the sign of the length, width, and height. Users wanting this behavior now will need to specify `L`, `W`, and `H` when constructing the cuboid, and then _again_ when calling `translate(-L/2, -W/2, -H/2)`. Or, some library will need to be introduced that allows choosing which corner to align to, and I don't see that being intuitive.

In short, I think most users expect the old behavior. For users who want the new behavior, it's as simple as calling center(). For those that want the old behavior, it's a lot more complicated.

**Comment by z3dev** [comment]
@hjfreyer your arguments seem sound, however we wanted to be 'consistant' across the APIs. And that's the main reason for centering of all primitives.

Consider the 'center' as part of the definition of the shape. Circles are "a shape consisting of all points in a plane that are a given distance from a given point, the center". And this is true about centers of cuboids as well due to the internal calculations of the points.

Also keep in mind that these are the 'core' functions, and it's very easy to mimick the old APIs. We want and encourage others to create various layers above these 'core' functions.

**Comment by z3dev** [comment]
Another good reason to center shapes is rotations. If a shape needs to be rotated then it must be oriented at 0,0,0, then translated into position.

**Comment by z3dev** [comment]
@hjfreyer there’s a new transformer function called align() which can align shapes, and place those aligned to a given pint.

This should make aligning shapes to a ‘corner’ very easy.

And center() should be rewritten to use align() as well.
